protocol/lavasession/provider_types.go

```diff
@@ -182,25 +182,26 @@ func (pswc *ProviderSessionsWithConsumer) atomicWriteMissingComputeUnits(cu uint
 	atomic.StoreUint64(&pswc.epochData.MissingComputeUnits, cu)
 }
 
-func (pswc *ProviderSessionsWithConsumer) SafeAddMissingComputeUnits(currentMissingCU uint64, allowedThreshold float64) (legitimate bool) {
+func (pswc *ProviderSessionsWithConsumer) SafeAddMissingComputeUnits(currentMissingCU uint64, allowedThreshold float64) (legitimate bool, totalMissingCu uint64) {
 	missing := pswc.atomicReadMissingComputeUnits()
 	used := pswc.atomicReadUsedComputeUnits()
 	max := pswc.atomicReadMaxComputeUnits()
+	totalMissingCu = missing + currentMissingCU
 	// do not allow bypassing max used CU
 	if currentMissingCU+missing+used > max {
-		return false
+		return false, totalMissingCu
 	}
 	// do not allow having more missing than threshold
 	if currentMissingCU+missing > uint64(float64(max)*allowedThreshold) {
-		return false
+		return false, totalMissingCu
 	}
 	// do not allow having more missing than already used
 	if currentMissingCU+missing > used {
-		return false
+		return false, totalMissingCu
 	}
 	// TODO: use compare and swap for race avoidance
```

Comment:
can you do TODO? it might be a race here that triggers this

Reply:
done [fa53d82](https://github.com/lavanet/lava/pull/586/commits/fa53d82e8f100b1ed50bd52dcee6fa1e5635f5e5)